corehq/apps/linked_domain/tasks.py

```diff
@@ -1,10 +1,123 @@
+import textwrap
+
 from celery.task import task
+from collections import defaultdict
+
+from django.conf import settings
+from django.utils.translation import ugettext as _
+
+from dimagi.utils.logging import notify_exception
 
-from corehq.apps.linked_domain.util import (
-    pull_missing_multimedia_for_app_and_notify,
-)
+from corehq import toggles
+from corehq.apps.app_manager.dbaccessors import get_apps_in_domain
+from corehq.apps.app_manager.util import is_linked_app
+from corehq.apps.app_manager.views.utils import update_linked_app
+from corehq.apps.hqwebapp.tasks import send_mail_async
+from corehq.apps.linked_domain.const import MODEL_APP, MODEL_CASE_SEARCH, MODEL_REPORT
+from corehq.apps.linked_domain.dbaccessors import get_linked_domains
+from corehq.apps.linked_domain.ucr import update_linked_ucr
+from corehq.apps.linked_domain.updates import update_model_type
+from corehq.apps.linked_domain.util import pull_missing_multimedia_for_app_and_notify
+from corehq.apps.userreports.dbaccessors import get_report_configs_for_domain
+from corehq.apps.users.models import CouchUser
 
 
 @task(queue='background_queue')
 def pull_missing_multimedia_for_app_and_notify_task(domain, app_id, email=None):
     pull_missing_multimedia_for_app_and_notify(domain, app_id, email)
+
+
+@task(queue='background_queue')
+def push_models(master_domain, models, linked_domains, build_apps, username):
```

Comment:
wow, this task is pretty complex - could it be broken up into separate pieces?

Reply:
yeah, working on it in https://github.com/dimagi/commcare-hq/pull/27878